vpr/src/base/vpr_types.h

```diff
@@ -795,6 +797,40 @@ struct hash<t_pl_loc> {
 };
 } // namespace std
 
+struct t_pl_atom_loc {
+    t_pl_atom_loc() = default;
+    t_pl_atom_loc(int primitive_id_, int x_, int y_, int sub_tile_, int layer_)
+        : primitive_id(primitive_id_)
+        , x(x_)
+        , y(y_)
+        , sub_tile(sub_tile_)
+        , layer(layer_) {}
+
+    int primitive_id = OPEN;
+    int x = OPEN;
+    int y = OPEN;
+    int sub_tile = OPEN;
+    int layer = OPEN;
+
+    friend bool operator==(const t_pl_atom_loc& lhs, const t_pl_atom_loc& rhs) {
```

Comment:
This struct has no private members, therefore it does not need to be given the friend attribute. I think it should instead be given the "inline" attribute unless you plan on adding private members in the future. See the following stack overflow post for more information: https://stackoverflow.com/questions/54197607/whats-the-use-of-a-friend-function-in-a-struct

Reply:
Note: I made it inline since you declared the operator overflow from within a header file instead of an implementation file.